service/health/searcher/types.go

```diff
@@ -47,3 +50,24 @@ func NewClusterStatus(cluster clusterInfo) ClusterStatus {
 		NodeCount: currentNodes,
 	}
 }
+
+func NewNodeStatus(node v1.Node) NodeStatus {
+	ready := false
+	for _, condition := range node.Status.Conditions {
+		if condition.Type == v1.NodeReady {
+			ready = condition.Status == "True"
```

Comment:
Double-checking this logic - is it the case that any "Ready" condition in this list makes the node ready, despite other conditions?

Reply:
Based on https://kubernetes.io/docs/concepts/architecture/nodes/#condition , I think this is the case. I haven't experienced enough unhealthy nodes to say for sure. Do you know @MarcelMue?

Reply:
This is explicitly checking for the `Ready` condition on the node - which defines readiness. It is not checking for all conditions in the loop. Therefor this should be fine.